src/dmd/expressionsem.d

```diff
@@ -8032,7 +8032,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 Expression el = new ArrayLengthExp(exp.loc, exp.e1);
                 el = el.expressionSemantic(sc);
                 el = el.optimize(WANTvalue);
-                if (el.op == EXP.int64)
+                if (el.op == EXP.int64 && t1b.ty == Tsarray)
```

Comment:
The length check should be performed only for static arrays, not dynamic arrays.

Reply:
This is only workaround for the specific case in the bug report. Static arrays are also affected by the regression (and not fixed by this patch):

```d
static template Bug(T, const T name)
{
    enum bool ok = name.length < 3 || name[0..3] != "pad";
}

pragma(msg, Bug!(char[1], "x").ok);
```